core/src/worker/mod.rs

```diff
@@ -78,8 +79,9 @@ use {
 pub struct Worker {
     config: WorkerConfig,
     wf_client: Arc<dyn WorkerClient>,
+    slot_provider: Arc<SlotProvider>,
     /// Registration key to enable eager workflow start for this worker
```

Comment:
@cretz Sorry for being so late in the game (already merged...) I wonder if holding a reference to slot_provider will hang the worker shutdown in some cases. slot_provider contains a reference to `external_wft_tx` and I think we need to drop all refs to it, so that the rx part closes, and that unblocks the worker shutdown...

Reply:
Wouldn't this have failed tests that rely on worker shutdown to complete (or are there none)? Is the `SlotProvider` drop needed in time for `shutdown` or `finalize_shutdown`. The latter consumes this worker and therefore this slot provider should be dropped.

Can we write a test for this? Do you have alternative suggestions on how to handle this?

Reply:
Ok, I have opened #725. I think that is a cleaner approach anyways and I wish I thought of it before.